### PR TITLE
Puppet 5 Modules for Stream Processor 4.2.0

### PR DESCRIPTION
**Pull request by irunika**
> Puppet version:  5.0

### CODE REVIEW

File: modules/sp_manager/manifests/init.pp [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+  # Copy the installer to the directory
+  file { "/opt/${product_name}/${sp_package}":
+    owner  => $user,
+    group  => $user_group,
+    mode   => '0644',
+    source => "puppet:///modules/${module_name}/${sp_package}",
+  }
+
+  # Install WSO2 API Manager
```
Comment (VimukthiPerera) [inline modules/sp_manager/manifests/init.pp]: Again "API Manager".

File: modules/sp_editor/manifests/params.pp [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  $carbon_db_password = 'wso2carbon'
+  $carbon_db_driver = 'org.h2.Driver'
+
+  # Cluster Configuration
+  $cluster_enabled = 'false'
+
+  #Authentication Configurations
+  $rest_api_auth_enable = 'false'
+
+
```
Comment (VimukthiPerera) [inline modules/sp_editor/manifests/params.pp]: Why do we have five extra empty lines here?

File: modules/sp_manager/manifests/params.pp [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  # Datasource Configurations
+  $mgt_db_url = 'jdbc:h2:${sys:carbon.home}/wso2/${sys:wso2.runtime}/database/SP_MGT_DB;DB_CLOSE_ON_EXIT=FALSE;LOCK_TIMEOUT=60000'
+  $mgt_db_username = 'wso2carbon'
+  $mgt_db_password = 'wso2carbon'
+  $mgt_db_dirver = 'org.h2.Driver'
+
+  # Cluster Configuration
+  $cluster_enabled = 'false'
+
+
```
Comment (VimukthiPerera) [inline modules/sp_manager/manifests/params.pp]: Multiple extra empty lines here as well.

File: modules/sp_worker/templates/wso2sp-worker.service.erb [hunk 4]
```diff
@@ -0,0 +1,19 @@
+[Unit]
+Description=WSO2 Stream Processor - Manager
```
Comment (VimukthiPerera) [inline modules/sp_worker/templates/wso2sp-worker.service.erb]: Worker service should have the Description=WSO2 Stream Processor - "Worker"

File: modules/sp_dashboard/manifests/init.pp [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+  $status_dashboard_db_password = $sp_dashboard::params::status_dashboard_db_password,
+  $status_dashboard_db_driver = $sp_dashboard::params::status_dashboard_db_driver,
+
+  $metrics_db_url = $sp_dashboard::params::metrics_db_url,
+  $metrics_db_username = $sp_dashboard::params::metrics_db_username,
+  $metrics_db_password = $sp_dashboard::params::metrics_db_password,
+  $metrics_db_driver = $sp_dashboard::params::metrics_db_driver,
+
+  # wso2.business.rules.manager
+  $businnes_rules_manager_username = $sp_dashboard::params::businnes_rules_manager_username,
```
Comment (DilanUA) [inline modules/sp_dashboard/manifests/init.pp]: minor spelling issue

File: modules/sp_dashboard/manifests/params.pp [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+  $status_dashboard_db_password = 'wso2carbon'
+  $status_dashboard_db_driver = 'org.h2.Driver'
+
+  $metrics_db_url = 'jdbc:h2:${sys:carbon.home}/wso2/dashboard/database/metrics;AUTO_SERVER=TRUE'
+  $metrics_db_username = 'wso2carbon'
+  $metrics_db_password = 'wso2carbon'
+  $metrics_db_driver = 'org.h2.Driver'
+
+  # wso2.business.rules.manager
+  $businnes_rules_manager_username = 'admin'
```
Comment (DilanUA) [inline modules/sp_dashboard/manifests/params.pp]: minor spelling issue

**Pull request merged**